.travis.yml

```diff
@@ -171,10 +171,16 @@ before_install:
         fi
   - virtualenv --version
   - python --version
+  - |
+        # upgrade setuptools if it's too old to support PEP 508
+        if [ $(pip list 2>/dev/null | grep setuptools | grep -oP '\d+' | head -n 1) -lt 21 ]; then
```

Comment:
Instead of doing this (seems rather complex), could you just check if setuptools version is >= 20.2? Also, please explain in the comment why do we need setuptools to support PEP-508. This way, when one day we no longer need that due to refactor or something, we'll know that we can remove it. (well, we can guess, because between now and then another reason to have it in this version may appear ;) )

In the future, could you please surround `$(...)` with double quotes `"$(...)"? In this project we have a policy to quote all $expressions. Probably should be mentioned somewhere in developer info in `README.md`...